sql/plan/group_by.go

```diff
@@ -420,7 +440,7 @@ func newAggregationBuffer(expr sql.Expression) (sql.AggregationBuffer, error) {
 		return n.NewBuffer()
 	default:
 		// The semantics for a non-aggregation in a group by node is Last.
```

Comment:
Might be nice to leave a note/clue about `ONLY_FULL_GROUP_BY` to help future readers understand this. 

```suggestion
		// The semantics for a non-aggregation expression in a group by node is First.
		// When ONLY_FULL_GROUP_BY is enabled, this is an error, but it's allowed otherwise. 
```